server/client.go

```diff
@@ -756,28 +756,36 @@ func (c *client) processErr(errStr string) {
 }
 
 // Password pattern matcher.
-var passPat = regexp.MustCompile(`"?\s*pass\S*?"?[:=]\s*"?(([^"])*)`)
+var passPat = regexp.MustCompile(`"?\s*pass\S*?"?\s*[:=]\s*"?(([^",\r\n}])*)`)
 
-// This will remove any notion of passwords from trace messages
-// for logging.
+// removePassFromTrace removes any notion of passwords from trace
+// messages for logging.
 func removePassFromTrace(arg []byte) []byte {
-
-	if !bytes.Contains(arg, []byte("pass")) {
+	if !bytes.Contains(arg, []byte(`pass`)) {
 		return arg
 	}
-	m := passPat.FindAllSubmatch(arg, -1)
+	// Take a copy of the connect proto just for the trace message.
+	a := make([]byte, len(arg))
```

Comment:
Maybe place on the stack since it probably does not escape? Compiler may do this for you these days.

I used to do
```
var _raw [4096]byte
buf := _raw[:0]
buf = append(buf, ...)
```

Reply:
Thanks, updated to take this approach.